vllm_spyre/v1/core/scheduler.py

```diff
@@ -140,6 +140,43 @@ def get_num_unfinished_requests(self) -> int:
         # Override this to include our extra queue
         return len(self.waiting) + len(self.running) + len(self.holdback_queue)
 
+    def finish_requests(
+        self,
+        request_ids: Union[str, Iterable[str]],
+        finished_status: RequestStatus,
+    ) -> None:
+        """Handles the finish signal from outside the scheduler.
+
+        For example, the API server can abort a request when the client
+        disconnects.
+
+        Specialized in vllm_spyre to handle the holdback_queue.
+        """
+        assert RequestStatus.is_finished(finished_status)
+        if isinstance(request_ids, str):
+            request_ids = (request_ids, )
+        else:
+            request_ids = set(request_ids)
+
+        for req_id in request_ids:
+            request = self.requests.get(req_id)
+            if request is None:
+                # Invalid request ID.
+                continue
+
+            if request.status == RequestStatus.RUNNING:
+                self.running.remove(request)
+                self.scheduled_req_ids.discard(request.request_id)
+            else:
+                # this try-except is the specialization for Spyre
+                try:
+                    self.holdback_queue.remove(request)
```

Comment:
Ah dang, this is unfortunate.

I think maybe we can fix this in a simpler way by removing `self.holdback_queue` as an instance attribute, and instead just make it a local variable during `self.schedule()`. After we schedule a new batch, we can take all the requests that we held back and put them back in `self.waiting`, and then we won't need to worry about breaking assumptions that the v1 scheduler has

Reply:
That would let us get rid of the override on `get_num_unfinished_requests` as well

Reply:
Done!
I left it as an instance variable so that we don't remake the deque. It is also still used in `_handle_rejects`, though there probably can't be rejection during scheduling?

Reply:
Right, I think that it should be safe to remove it from usage in `_handle_rejects` as well since this should all be synchronous. The output processing that calls `_handle_rejects` can't be happening concurrently with scheduling a new forward pass

Reply:
I'll leave it up to you on merging now vs. pulling out of _handle_rejects as well. I think we'll be getting rid of this rejected request business soon anyway